lightning/src/routing/router.rs

```diff
@@ -373,11 +373,14 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 	if final_value_msat == 0 {
 		return Err(LightningError{err: "Cannot send a payment of 0 msat".to_owned(), action: ErrorAction::IgnoreError});
 	}
+	
+	let last_hops = &(*last_hops).clone().iter().map(|hops| hops.0.clone()).collect::<Vec<_>>();
```

Comment:
You should be able to avoid the clone entirely, I think, and just do `last_hops.iter()`.